### PR TITLE
Upstream downstream pop ups

### PR DESCRIPTION
**Pull request by Tehsurfer**
image

Branch can be view at:
https://context-cards-demo.herokuapp.com/maps

This is for this milestone: 
https://www.wrike.com/open.htm?id=685913854

### CODE REVIEW

File: src/App.vue [hunk 1]
```diff
@@ -91,8 +91,10 @@ export default {
       displayCloseButton: false,
       initial: "Rat",
       helpMode: false,
+      // flatmapAPI: "https://mapcore-demo.org/current/flatmap/v2/"
+      flatmapAPI: "https://mapcore-demo.org/devel/flatmap/v3/"
```
Comment (alan-wu) [inline src/App.vue:95]: Is the v3 for testing or required? What happen if v1 (current production) is used?

**Review by alan-wu — CHANGES_REQUESTED** [review]
Looks fine but need to consider keeping the API location consistent for testing. Ideally they should all be pointing to the same one.

I would like to see the semicolon to be more consistent.  ( I still believe semi colon should be used for the rare cases that ASI triggers unintentionally.)

File: src/App.vue (hunk 1 above)
Comment (Tehsurfer) [reply src/App.vue:95]: Yeah, v3 is needed, the previous maps are not annotated with `soma` and `axons` which I need for the tooltip.

if v2 is used, the tooltips get stuck while loading. I will put an update in to fix this

File: src/components/Tooltip.vue [hunk 2]
```diff
@@ -72,36 +98,168 @@ export default {
...
+            found.push(node)
+          }
+      })
+
+      // remove duplicates
+      let foundUnique = [...new Set(found.map(n=>n[0]))]
+      return foundUnique
+    },
+    getOrganCuries: function(){
+      fetch('https://api.sparc.science/get-organ-curies/')
```
[the remark this answers is not in the document]
Comment (Tehsurfer) [reply src/components/Tooltip.vue:180]: yeah I just was being lazy and didn't want to switch flatmapvuer to the envVars prop but I guess I will have to?

File: src/components/Tooltip.vue [hunk 3]
```diff
@@ -54,16 +70,26 @@ Vue.use(Header);
...
+  return str.charAt(0).toUpperCase() + str.slice(1)
+}
+
 export default {
   components: { PubmedViewer },
   name: "Tooltip",
   props: { 
+    flatmapAPI: {
+      type: String,
+      default: 'https://mapcore-demo.org/current/flatmap/v2/'
```
[the remark this answers is not in the document]
Comment (Tehsurfer) [reply src/components/Tooltip.vue:91]: > This should be using the endpoint provided to the flatmapvuer.
it is using the endpoint from flatmapvuer isn't it? 

> We can potentially use Provide / Inject to use props several layers deep.
Ok I will have a look at this

**Comment by Tehsurfer** [comment]
> Looks fine but need to consider keeping the API location consistent for testing. Ideally they should all be pointing to the same one.
> 
> I would like to see the semicolon to be more consistent. ( I still believe semi colon should be used for the rare cases that ASI triggers unintentionally.)

We can keep the api consistent, I had it there mostly so you knew which one was being used

File: src/components/Tooltip.vue (hunk 2 above)
[the remark this answers is not in the document]
Comment (alan-wu) [reply src/components/Tooltip.vue:180]: Yes, we could be testing other instances of API.

File: src/components/Tooltip.vue (hunk 3 above)
[the remark this answers is not in the document]
Comment (alan-wu) [reply src/components/Tooltip.vue]: Yes, it is. Sorry, it is the default value of a prop here.

**Comment by Tehsurfer** [comment]
> @Tehsurfer Looks good, but the Search for dataset button is not working this moment. Has that been fixed on the sidebar? Why does it take multiple clicks for the dialog to disappear?

I've commented out the 'search for datasets' button. There aren't any keastids on scicrunch anyways. I've switched it to a "Search on components" button, which puts all components as a facet.

I looked deeper into the multiple clicks and any time I use         
```
this.mapImp.showPopup(
          this.mapImp.modelFeatureIds(data.resource[0])[0],
          'hello!',
        );
```
it takes two clicks to close. Not sure why but the popup from the flatmap seems to be 'stealing' the first click

File: src/components/Tooltip.vue (hunk 2 above)
[the remark this answers is not in the document]
Comment (Tehsurfer) [reply src/components/Tooltip.vue]: now fixed

**Review by alan-wu — COMMENTED** [review]
Looks good to me. Can you remove the multi flatmapQueries.js please?

**Review by alan-wu — APPROVED** [review]

**Pull request merged**